test/battle/ability/commander.c

```diff
@@ -0,0 +1,440 @@
+    } THEN {
+        switch (species)
+        {
+        case SPECIES_TATSUGIRI_CURLY:
+            EXPECT_EQ(playerRight->statStages[STAT_ATK], DEFAULT_STAT_STAGE + 1);
+            break;
+        case SPECIES_TATSUGIRI_DROOPY:
+            EXPECT_EQ(playerRight->statStages[STAT_DEF], DEFAULT_STAT_STAGE + 1);
+            break;
+        case SPECIES_TATSUGIRI_STRETCHY:
+            EXPECT_EQ(playerRight->statStages[STAT_SPEED], DEFAULT_STAT_STAGE + 1);
+            break;
+        }
+    }
+}
+
+DOUBLE_BATTLE_TEST("(Commander) Whirlwind can not be used against Dondozo or Tatsugiri while Commander is active")
+{
+    GIVEN {
+        PLAYER(SPECIES_TATSUGIRI) { Ability(ABILITY_COMMANDER); }
+        PLAYER(SPECIES_DONDOZO);
+        PLAYER(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(opponentLeft, MOVE_WHIRLWIND, target: playerLeft); }
+        TURN { MOVE(opponentRight, MOVE_WHIRLWIND, target: playerRight); }
+    } SCENE {
+        ABILITY_POPUP(playerLeft, ABILITY_COMMANDER);
+        MESSAGE("Tatsugiri was swallowed by Dondozo and became Dondozo's commander!");
+        MESSAGE("Foe Wobbuffet used Whirlwind!");
+        MESSAGE("But it failed!");
+        MESSAGE("Foe Wobbuffet used Whirlwind!");
+        MESSAGE("But it failed!");
+    }
+}
+
+DOUBLE_BATTLE_TEST("(Commander) Red Card fails on Dondozo while Commander is active")
+{
+    GIVEN {
+        PLAYER(SPECIES_TATSUGIRI) { Ability(ABILITY_COMMANDER); }
+        PLAYER(SPECIES_DONDOZO);
+        PLAYER(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET) { Item(ITEM_RED_CARD); }
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(playerRight, MOVE_TACKLE, target: opponentLeft); }
+    } SCENE {
+        ABILITY_POPUP(playerLeft, ABILITY_COMMANDER);
+        MESSAGE("Tatsugiri was swallowed by Dondozo and became Dondozo's commander!");
+        ANIMATION(ANIM_TYPE_GENERAL, B_ANIM_HELD_ITEM_EFFECT, opponentLeft);
+    }
+}
+
+DOUBLE_BATTLE_TEST("(Commander) Tatsugiri is not damaged by a double target move if Dondozo faints")
+{
+    KNOWN_FAILING;
+    /*
+    test/battle/ability/commander.c:260:
+    1 TURNs specified, but 2 ran - (Commander) Tatsugiri is not damaged by a double target move if Dondozo faints.
+    */
+    GIVEN {
+        ASSUME(gMovesInfo[MOVE_SURF].target == MOVE_TARGET_FOES_AND_ALLY);
+        PLAYER(SPECIES_DONDOZO) { HP(1); };
+        PLAYER(SPECIES_TATSUGIRI) { Ability(ABILITY_COMMANDER); }
+        OPPONENT(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(playerLeft, MOVE_CELEBRATE); MOVE(opponentLeft, MOVE_SURF); }
+    } SCENE {
+        ABILITY_POPUP(playerRight, ABILITY_COMMANDER);
+        MESSAGE("Tatsugiri was swallowed by Dondozo and became Dondozo's commander!");
+        HP_BAR(playerLeft);
+        MESSAGE("Dondozo fainted!");
+        HP_BAR(opponentRight);
+        NOT HP_BAR(playerRight);
```

Comment:
I believe the order on one of the `HP_BAR`s is incorrect which is why it asks for a second turn

Reply:
Test still fails but with the correct failing message now